### PR TITLE
[SPARK-45868][CONNECT] Make sure `spark.table` use the same parser with vanilla spark

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Make sure spark.table use the same parser with vanilla spark 

### Why are the changes needed?
to be consistent with the vanilla spark:

https://github.com/apache/spark/blob/9d93b7112a31965447a34301889f90d14578e628/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L714-L720

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Review by ueshin — APPROVED** [review]
LGTM. Thanks for the fix!

**Review by dongjoon-hyun — COMMENTED** [review]
Can we have a test case for this, @zhengruifeng ?

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 1]
```diff
@@ -1217,7 +1217,8 @@ class SparkConnectPlanner(
     rel.getReadTypeCase match {
       case proto.Read.ReadTypeCase.NAMED_TABLE =>
         val multipartIdentifier =
-          CatalystSqlParser.parseMultipartIdentifier(rel.getNamedTable.getUnparsedIdentifier)
+          session.sessionState.sqlParser
+            .parseMultipartIdentifier(rel.getNamedTable.getUnparsedIdentifier)
```
Comment (HyukjinKwon) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:1221]: I think we can add a test under `SparkConnectPlannerSuite` that uses a cloned `SparkSession` that defines a separate parser with `SparkSession.withExtensions` that overrides `parseMultipartIdentifier` which throws an exception (see `MyParser` at `SparkSessionExtensionSuite`).

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise.

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala (hunk 1 above)
Comment (zhengruifeng) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:1221]: thanks, I will have a try

Comment (zhengruifeng) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:1222]: done

**Comment by zhengruifeng** [comment]
since it dose cause some issues in productions, I think we need to also backport it to 3.5

**Comment by zhengruifeng** [comment]
cc @dongjoon-hyun @HyukjinKwon would you mind taking another look?

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs). Thank you for updates, @zhengruifeng .

**Review by beliefer — APPROVED** [review]
LGTM.

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @zhengruifeng and all!

**Pull request closed without merging**